drivers/flash/spi_nor.c

```diff
@@ -21,6 +19,14 @@
 #include "jesd216.h"
 #include "flash_priv.h"
 
+#if DT_HAS_COMPAT_STATUS_OKAY(jedec_spi_nor)
+#define DT_DRV_COMPAT jedec_spi_nor
+#elif DT_HAS_COMPAT_STATUS_OKAY(mxicy_mx25r_spi)
+#define DT_DRV_COMPAT mxicy_mx25r_spi
+#else
+#error No known devicetree compatible match for SPI NOR driver
+#endif
```

Comment:
Do we even have that compatible? The param `mxicy,high-perf-mode` is already mx  characteristic, so we probably only should check if the device is mx and does have the capability, when `spi-nor` compatible device is given the param in DTS.

Reply:
But isn't `DT_DRV_COMPAT` used for the `DEVICE_DT_INST_DEFINE` at the bottom of `spi_nor.c`, and perhaps other macro functions used in the file?

Reply:
It is. And `"jedec,spi-nor"` will work fine in all those cases.
Perhaps I wasn't clear in my comment. By writing "no need to change `DT_DRV_COMPAT` here" I meant keeping `#define DT_DRV_COMPAT jedec_spi_nor` as it was so far at the beginning of this file and removing the part cited above.

Reply:
According to the [dts bindings docs](https://docs.zephyrproject.org/latest/build/dts/bindings.html#other-ways-nodes-are-matched-to-bindings): «If a node has more than one string in its compatible property, the build system looks for compatible bindings in the listed order and uses the first match.»
So if we have `DT_DRV_COMPAT` always use `jedec_spi_nor` we would be mixing binding for the same device tree node when `mxicy,mx25r-spi` is supposed to be used.

The suggested method from @mbolivar was to use `DT_DRV_COMPAT` for one compatible and loop through devices that match the compatible with `DT_INST_FOREACH_STATUS_OKAY`. Then undef `DT_DRV_COMPAT` and define it for the next compatible and loop through those. Like is done here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/tmp108/tmp108.c#L406

However that requires modifying the driver to support multiple device instances, and I don't want to do that in the same PR as that could introduce other issues and the whole PR could be reverted. So I think this can be a future improvement in the driver.

The method to check for an enabled node with a compatible before defining `DT_DRV_COMPAT` [is used in many drivers](https://github.com/zephyrproject-rtos/zephyr/search?q=%23elif+DT_HAS_COMPAT_STATUS_OKAY).  Is there a problem of using this method? To me this seems like a cleaner solution when the driver only support one device instance.